packages/model-serving/src/components/deploymentWizard/fields/NumReplicasField.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { FormGroup, HelperText, HelperTextItem } from '@patternfly/react-core';
+import { z } from 'zod';
+import NumberInputWrapper from '@odh-dashboard/internal/components/NumberInputWrapper';
+import { normalizeBetween } from '@odh-dashboard/internal/utilities/utils';
+
+// Schema
+export const numReplicasFieldSchema = z.number().min(1).max(99);
+
+export type NumReplicasFieldData = z.infer<typeof numReplicasFieldSchema>;
+
+export const isValidNumReplicas = (value: unknown): value is NumReplicasFieldData => {
+  return numReplicasFieldSchema.safeParse(value).success;
+};
+
+// Hook
+export type NumReplicasFieldHook = {
+  data: NumReplicasFieldData | undefined;
+  setReplicas: (replicas: number) => void;
+};
+
+export const useNumReplicasField = (existingData?: NumReplicasFieldData): NumReplicasFieldHook => {
+  const [replicaData, setReplicaData] = React.useState<NumReplicasFieldData | undefined>(
+    existingData || 1,
+  );
+
+  const setReplicas = React.useCallback((replicas: number) => {
+    setReplicaData(replicas);
+  }, []);
+
+  return {
+    data: replicaData,
+    setReplicas,
+  };
+};
+
+// Component
+type NumReplicasFieldProps = {
+  replicaState: NumReplicasFieldHook;
+};
+
+const lowerLimit = 1;
+const upperLimit = 99;
```

Comment:
I would set these at the top of the file and use all caps like `LOWER_LIMIT` and `UPPER_LIMIT`. Then replace anywhere we have `1` or `99`